cli/cmd/generate_oci.go

```diff
@@ -0,0 +1,437 @@
+	}
+
+	return nil
+}
+
+func preRunGenerateOci(cmd *cobra.Command, _ []string) error {
+	// Validate output location is OK if supplied
+	dirname, err := cmd.Flags().GetString("output")
+	if err != nil {
+		return errors.Wrap(err, "failed to load command flags")
+	}
+	if err := validateOutputLocation(dirname); err != nil {
+		return err
+	}
+
+	// Validate tenant OCID
+	tenantOcid, err := cmd.Flags().GetString("tenant_ocid")
+	if err != nil {
+		return errors.Wrap(err, "failed to load command flags")
+	}
+	if err := validateOciTenantOcid(tenantOcid); tenantOcid != "" && err != nil {
+		return err
+	}
+
+	// Validate user email
+	ociUserEmail, err := cmd.Flags().GetString("oci_user_email")
+	if err != nil {
+		return errors.Wrap(err, "failed to load command flags")
+	}
+	if err := validateOciUserEmail(ociUserEmail); ociUserEmail != "" && err != nil {
+		return err
+	}
+
+	// Load any cached inputs if interactive
+	if cli.InteractiveMode() {
+		cachedOptions := &oci.GenerateOciTfConfigurationArgs{}
+		iacParamsExpired := cli.ReadCachedAsset(CachedOciAssetIacParams, &cachedOptions)
+		if iacParamsExpired {
+			cli.Log.Debug("loaded previously set values for OCI iac generation")
+		}
+
+		extraState := &OciGenerateCommandExtraState{}
+		extraStateParamsExpired := cli.ReadCachedAsset(CachedAssetOciExtraState, &extraState)
+		if extraStateParamsExpired {
+			cli.Log.Debug("loaded previously set values for OCI iac generation (extra state)")
+		}
+
+		// Determine if previously cached options exists; prompt user if they'd like to continue
+		answer := false
+		if !iacParamsExpired || !extraStateParamsExpired {
+			if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+				Prompt:   &survey.Confirm{Message: QuestionUsePreviousCache, Default: false},
+				Response: &answer,
+			}); err != nil {
+				return errors.Wrap(err, "failed to load saved options")
+			}
+		}
+
+		// If the user decides NOT to use the previous values; we won't load them.  However, every time the command runs
+		// we are going to write out new cached values, so if they run it - bail out - and run it again they'll get
+		// re-prompted.
+		if answer {
+			// Merge cached inputs to current options (current options win)
+			if err := mergo.Merge(GenerateOciCommandState, cachedOptions); err != nil {
+				return errors.Wrap(err, "failed to load saved options")
+			}
+			if err := mergo.Merge(GenerateOciCommandExtraState, extraState); err != nil {
+				return errors.Wrap(err, "failed to load saved options")
+			}
+		}
+	}
+
+	// Collect and/or confirm parameters
+	err = promptOciGenerate(GenerateOciCommandState, GenerateOciCommandExtraState)
+	if err != nil {
+		return errors.Wrap(err, "collecting/confirming parameters")
+	}
+
+	return nil
+}
+
+func initGenerateOciTfCommandFlags() {
+	// add flags to sub commands
+	generateOciTfCommand.PersistentFlags().BoolVar(
+		&GenerateOciCommandState.Config,
+		"config",
+		false,
```

Comment:
Why have this as an option if it is mandatory?

Reply:
I was just trying to make the flow of questions more consistent with the other three providers. Based on the general feedback I've gotten, it sounds like it would be best to skip this option/question.

Reply:
Leaving as is for now